src/identity/did-document.ts

```diff
@@ -15,10 +15,38 @@ import { HcsDidCreateVerificationRelationshipEvent } from "./hcs/did/event/verif
 import { HcsDidRevokeVerificationRelationshipEvent } from "./hcs/did/event/verification-relationship/hcs-did-revoke-verification-relationship-event";
 import { HcsDidUpdateVerificationRelationshipEvent } from "./hcs/did/event/verification-relationship/hcs-did-update-verification-relationship-event";
 
+export enum DidResolveError {
+    INVALID_DID = "invalidDid",
+    NOT_FOUND = "notFound",
+    REPRESENTATION_NOT_SUPPORTED = "representationNotSupported",
+}
+
 export class DidDocument {
     private readonly id: string;
     private readonly context: string;
 
+    /**
+     * TODO: at the moment error is no where to bet set.
+     * Probably more resolve logic should be moved here so it could be used independently from HcsDid class.
+     * This way error handling could be implemented here too.
+     */
```

Comment:
I was thinking the opposite. 

how about we move this logic to hcs-did as 
this DID resolution result metadata is only applicable to 
Resolve. 

This class is responsible for producing did-document.